acl/src/main/java/org/apache/rocketmq/acl/plain/PlainPermissionManager.java

```diff
@@ -393,6 +399,401 @@ public boolean updateAccessConfig(PlainAccessConfig plainAccessConfig) {
+                    if (entry.getValue().getAccessKey().equals(plainAccessConfig.getAccessKey())) {
+                        accountMap.put(entry.getKey(), updatePlainAccessResource);
+                        break;
+                    }
+                }
+            }
+            aclPlainAccessResourceMap.put(aclFileName, accountMap);
+            return AclUtils.writeDataObject(aclFileName, updateAclConfigFileVersion(aclFileName, aclAccessConfigMap));
+        } else {
+            String fileName = MixAll.dealFilePath(defaultAclFile);
+            //Create acl access config elements on the default acl file
+            if (aclPlainAccessResourceMap.get(defaultAclFile) == null || aclPlainAccessResourceMap.get(defaultAclFile).size() == 0) {
+                try {
+                    File defaultAclFile = new File(fileName);
+                    if (!defaultAclFile.exists()) {
+                        defaultAclFile.createNewFile();
+                    }
+                } catch (IOException e) {
+                    log.warn("create default acl file has exception when update accessConfig. ", e);
+                }
+            }
+            Map<String, Object> aclAccessConfigMap = AclUtils.getYamlDataObject(defaultAclFile, Map.class);
+            if (aclAccessConfigMap == null) {
+                aclAccessConfigMap = new HashMap<>();
+                aclAccessConfigMap.put(AclConstants.CONFIG_ACCOUNTS, new ArrayList<>());
+            }
+            List<Map<String, Object>> accounts = (List<Map<String, Object>>) aclAccessConfigMap.get(AclConstants.CONFIG_ACCOUNTS);
+            // When no accounts defined
+            if (null == accounts) {
+                accounts = new ArrayList<>();
+            }
+            accounts.add(createAclAccessConfigMap(null, plainAccessConfig));
+            aclAccessConfigMap.put(AclConstants.CONFIG_ACCOUNTS, accounts);
+            accessKeyTable.put(plainAccessConfig.getAccessKey(), fileName);
+            if (aclPlainAccessResourceMap.get(fileName) == null) {
+                Map<String, PlainAccessResource> plainAccessResourceMap = new HashMap<>(1);
+                plainAccessResourceMap.put(plainAccessConfig.getAccessKey(), buildPlainAccessResource(plainAccessConfig));
+                aclPlainAccessResourceMap.put(fileName, plainAccessResourceMap);
+            } else {
+                Map<String, PlainAccessResource> plainAccessResourceMap = aclPlainAccessResourceMap.get(fileName);
+                plainAccessResourceMap.put(plainAccessConfig.getAccessKey(), buildPlainAccessResource(plainAccessConfig));
+                aclPlainAccessResourceMap.put(fileName, plainAccessResourceMap);
+            }
+            return AclUtils.writeDataObject(defaultAclFile, updateAclConfigFileVersion(defaultAclFile, aclAccessConfigMap));
+        }
+    }
+
+    public PlainAccessResource updatePlainAccessResource(Map<String, Object> updateAccountMap) {
```

Comment:
In this method, `PlainAccessResource` is **created** but not **updated.** The signature of methods should be more descriptive.
How about modifying `org.apache.rocketmq.acl.plain.PlainPermissionManager#buildPlainAccessResource` to support new config format?